internal/component/otelcol/receiver/prometheus/prometheus.go

```diff
@@ -117,11 +118,9 @@ func (c *Component) Update(newConfig component.Arguments) error {
 		gcInterval = 5 * time.Minute
 	)
 
-	cTypeStr := strings.ReplaceAll(strings.ReplaceAll(c.opts.ID, ".", "_"), "/", "_")
-
 	settings := otelreceiver.CreateSettings{
 
-		ID: otelcomponent.NewID(otelcomponent.MustNewType(cTypeStr)),
+		ID: otelcomponent.NewID(otelcomponent.MustNewType(typeStr)),
```

Comment:
In the other packages that use CreateSettings (like otelcol/exporter), they omit the ID entirely; should that be done here too? 

The ID is really only needed for extensions IIUC

Reply:
Yeah, let's go with that, I've removed the struct field entirely from here.